File: app/src/main/java/com/sample/android/moviebrowser/tasks/ReadDBMovieListTask.java
package com.sample.android.moviebrowser.tasks;

import java.util.ArrayList;

import com.sample.android.moviebrowser.MovieListActivity;
import com.sample.android.moviebrowser.db.DBManager;
import com.sample.android.moviebrowser.db.MovieManager;
import com.sample.android.moviebrowser.models.Movie;

import android.os.AsyncTask;

/**
 * @author Carlos Vasconcelos
 *
 */
public class ReadDBMovieListTask extends AsyncTask<Void, Void, Void> {
	private DBManager dbManager;
	private MovieListActivity activity;
	private ArrayList<Movie> movieList;
	
	
	public ReadDBMovieListTask(MovieListActivity activity, DBManager dbManager) {
		this.activity = activity;
		this.dbManager = dbManager;
		movieList = new ArrayList<Movie>();
	}

	
	@Override
	protected Void doInBackground(Void... params) {
		MovieManager movieManager = new MovieManager(activity, dbManager);
		movieList = movieManager.getMoviesList();
		
		return null;
	}

    protected void onPostExecute(Void result) {
    	activity.updateMovieList(movieList);
    }



}

File: app/src/main/java/com/sample/android/moviebrowser/tasks/ReadDBMovieListTask.kt
package com.sample.android.moviebrowser.tasks

import java.util.ArrayList

import com.sample.android.moviebrowser.MovieListActivity
import com.sample.android.moviebrowser.db.DBManager
import com.sample.android.moviebrowser.db.MovieManager
import com.sample.android.moviebrowser.models.Movie

import android.os.AsyncTask

/**
 * @author Carlos Vasconcelos
 */
class ReadDBMovieListTask(private val activity: MovieListActivity, private val dbManager: DBManager) : AsyncTask<Void, Void, Void>() {
    private var movieList: ArrayList<Movie>? = null


    init {
        movieList = ArrayList<Movie>()
    }


    override fun doInBackground(vararg params: Void): Void? {
        val movieManager = MovieManager(activity, dbManager)
        movieList = movieManager.moviesList

        return null
    }

    override fun onPostExecute(result: Void) {
        activity.updateMovieList(movieList!!)
    }


}
